### PR TITLE
Opengl Quick Fixes

### PR DESCRIPTION
**Pull request by alama**
Various quick fixes so OpenGL isn't *quite* so bad.

*  Some basic fixes for sorting between MD2s and sprites so they aren't quite so bad at times
*  Make walls/floors that have holes in, but aren't translucent, be drawn as solids, so they don't need to be sorted manually and use the depth buffer
*  Make texture filtering not do anything on textures or flats with holes in them (they use nearest) because the added translucency can break things. If correct sorting is implemented they could use it again but would need to be sorted manually
*  Decrease the far clipping plane from it's actually absurd value to a more reasonable 32768, which is probably as far as software can see anyway. Hopefully will help with depth buffer precision
*  Optimises the screen texture setups so that they aren't redefining textures every frame if they don't need to. This gives a pretty large performance increase, especially on Intel GPUs
*  Fix screenshots and fade wipes in SDL2 for resolutions that are lower than the monitor's native resolution
*  Slightly better Fog Block colouring. Still not particularly accurate, but better
*  Stop translucent floors/ceilings from being drawn into the depth buffer
*  Match the currently used OpenGL drawing functions to the software ones. There are a few others that don't fully match still, but they also aren't currently used from what I can see, so I have no point of reference for how they should work.
*  Make the screen texture be drawn in the centre of the screen with black bars in SDL2, instead of being stretched across the entire screen
*  Update and fix a bit HWR_SplitWall so it actually works with water properly where it wasn't before.
*  Sprite Splitting with a sector's light list in OpenGL (Needed to make sprites be transformed in world space rather than screen space so this worked easier, stransform has been removed since it ended up unused)
*  The geometry being rendered was being squashed because it was always assuming that a standard resolution was being used, this has been fixed
*  Comment out the call to HWR_CorrectSWTricks. I don't think it was being used, or if it was needed it wasn't being done properly probably. If it's not needed at all, all the parts around it can probably be removed
*  A really silly fix I still don't understand, but it fixes the lighting issue in DSZ3

### CODE REVIEW

File: src/f_finale.c [hunk 1]
```diff
@@ -233,11 +233,19 @@ static void F_SkyScroll(INT32 scrollspeed)
 #ifdef HWRENDER
 	else if (rendermode != render_none)
 	{ // if only software rendering could be this simple and retarded
-		scrolled = animtimer;
-		if (scrolled > 0)
-			V_DrawScaledPatch(scrolled - SHORT(pat->width), 0, 0, pat);
-		for (x = 0; x < fakedwidth; x += SHORT(pat->width))
-			V_DrawScaledPatch(x + scrolled, 0, 0, pat);
+		INT32 dupz = (vid.dupx < vid.dupy ? vid.dupx : vid.dupy);
+		INT32 x, y, pw = SHORT(pat->width) * dupz, ph = SHORT(pat->height) * dupz;
```
Comment (alama) [inline src/f_finale.c]:
```
f_finale.c: In function 'F_SkyScroll':
f_finale.c:237:11: error: declaration of 'x' shadows a previous local [-Werror=shadow]
   INT32 x, y, pw = SHORT(pat->width) * dupz, ph = SHORT(pat->height) * dupz;
           ^
```

File: src/hardware/hw_main.c [hunk 2]
```diff
@@ -4914,7 +5054,8 @@ static void HWR_CreateDrawNodes(void)
 //  Draw all vissprites
 // --------------------------------------------------------------------------
 #ifdef SORTING
-static void HWR_DrawSprites(void)
+// added the stransform so they can be switched as drawing happenes so MD2s and sprites are sorted correctly with each other
+static void HWR_DrawSprites()
```
Comment (alama) [inline src/hardware/hw_main.c]:
```
hardware/hw_main.c: In function 'HWR_DrawSprites':
hardware/hw_main.c:5058:13: error: old-style function definition [-Werror=old-style-definition]
 static void HWR_DrawSprites()
             ^
```

File: src/f_finale.c [hunk 3]
```diff
@@ -999,7 +1007,7 @@ static const char *credits[] = {
 	"",
 	"\1Sprite Artists",
 	"Odi \"Iceman404\" Atunzu",
-	"Victor \"VAdaPEGA\" Ara\x1Fjo", // Ara�jo -- sorry for our limited font! D:
+	"Victor \"VAdaPEGA\" Ara\x1Fjo", // Araújo -- sorry for our limited font! D:
```
Comment (alama) [inline src/f_finale.c]: What did you change this?

Comment (Sryder13) [reply src/f_finale.c]: For some reason, my editor didn't like the 'ú' character... I tried setting it back and it still thought I changed it. Not sure what happened there.

**Pull request merged**